templates/vm.xml.j2

```diff
@@ -21,20 +26,29 @@
   <devices>
     <emulator>{{ libvirt_vm_emulator }}</emulator>
 {% for volume in volumes %}
-    <disk type='volume' device='{{ volume.device | default(libvirt_volume_default_device) }}'>
+    <disk type='{{ volume.type |default(libvirt_volume_default_type) }}' device='{{ volume.device | default(libvirt_volume_default_device) }}'>
       <driver name='qemu' type='{{ volume.format | default(libvirt_volume_default_format) }}'/>
+      {% if volume.type is defined and volume.type == 'file' %}
+      <source file='/var/lib/libvirt/images/{{ volume.name}}.qcow2'/>
+      {% else %}
       <source pool='{{ volume.pool }}' volume='{{ volume.name }}'/>
```

Comment:
Do we need a default for the pool here also?

Reply:
I don't think this is required if disk type='file'

Reply:
https://libvirt.org/formatdomain.html#elementsDisks FWIW, none of the examples in the domain format show a pool for `source file`.